tests/backup/backup_test.go

```diff
@@ -61,18 +74,121 @@ func TearDownBackupRestore(bkpNamespaces []string, restoreNamespaces []string) {
 	DeleteCloudCredential(CredName, OrgID, CloudCredUID)
 }
 
-func DeleteCloudAccounts(backupLocationMap map[string]string, credName string, cloudCredUID string) {
-	for backupLocationUID, bkpLocationName := range backupLocationMap {
-		DeleteBackupLocation(bkpLocationName, backupLocationUID, orgID)
+func DeleteCloudAccounts(backupLocationMap map[string]string, credName string, cloudCredUID string, ctx context.Context) {
+	if len(backupLocationMap) != 0 {
+		for backupLocationUID, bkpLocationName := range backupLocationMap {
+			DeleteBackupLocation(bkpLocationName, backupLocationUID, orgID)
+		}
+		time.Sleep(time.Minute * 3)
+		DeleteCloudCredential(credName, orgID, cloudCredUID)
 	}
-	time.Sleep(time.Minute * 3)
-	ctx, err := backup.GetAdminCtxFromSecret()
-	log.FailOnError(err, "Fetching px-central-admin ctx")
-	DeleteCloudCredential(credName, orgID, cloudCredUID)
 	DeleteCluster(SourceClusterName, orgID, ctx)
 	DeleteCluster(destinationClusterName, orgID, ctx)
 }
 
+// AddRoleAndAccessToUsers assigns role and access level to the users
+// AddRoleAndAccessToUsers return map whose key is userRoleAccess and value is backup for each user
+func AddRoleAndAccessToUsers(users []string, backupNames []string) (map[userRoleAccess]string, error) {
```

Comment:
Can you move this function `AddRoleAndAccessToUsers` and also `ValidateSharedBackupWithUsers` to the bottom of the file since we have placed almost all the helper functions there.

Reply:
done